diff --git a/modules/geocluster_solr/includes/GeoclusterSearchApiSolrService.inc b/modules/geocluster_solr/includes/GeoclusterSearchApiSolrService.inc
index 947e917..3f2ad22 100644
--- a/modules/geocluster_solr/includes/GeoclusterSearchApiSolrService.inc
+++ b/modules/geocluster_solr/includes/GeoclusterSearchApiSolrService.inc
@@ -81,8 +81,9 @@ class GeoclusterSearchApiSolrService extends SearchApiSolrService {
    * @param Apache_Solr_Response $response
    *   The response object returned by Solr.
    */
-  protected function postQuery(array &$results, SearchApiQueryInterface $query, Apache_Solr_Response $response) {
+  protected function postQuery(array &$results, SearchApiQueryInterface $query, $response) {
     if (module_exists('devel')) {
+      timer_start('geocluster');
       dd(timer_read("geocluster") . "ms: started clustering");
     }
 
@@ -139,7 +140,7 @@ class GeoclusterSearchApiSolrService extends SearchApiSolrService {
         dd(timer_read("geocluster") . "ms: pre-clustered by geohash: " . count($groups));
       }
 
-      $this->algorithm->setValues(&$results['results']);
+      $this->algorithm->setValues($results['results']);
       $this->algorithm->clusterByNeighborCheck($results_by_geohash);
 
       if (module_exists('devel')) {
@@ -164,7 +165,7 @@ class GeoclusterSearchApiSolrService extends SearchApiSolrService {
    *   - results: An array of search results, as specified by
    *     SearchApiQueryInterface::execute().
    */
-  protected function extractResults(SearchApiQueryInterface $query, Apache_Solr_Response $response) {
+  protected function extractResults(SearchApiQueryInterface $query, $response) {
     if ($this->algorithm) {
       // Set grouping options, so search api will process single result groups correctly.
       $options = array(
